survey/src/main/java/com.quickbirdstudios.surveykit/backend/views/question_parts/DateTimePickerPart.kt

```diff
@@ -0,0 +1,130 @@
+            layoutParams.bottomMargin =
+                context.resources.getDimensionPixelSize(R.dimen.question_text_padding)
+            this.layoutParams = layoutParams
+            background = ContextCompat.getDrawable(context, R.drawable.main_button_background)
+        }
+        showDatePickerButton = Button(context).apply {
+            text = context.getString(R.string.date_time_picker_show_date_button_text)
+            setOnClickListener { datePicker.show() }
+            val layoutParams = LayoutParams(
+                LayoutParams.WRAP_CONTENT,
+                LayoutParams.WRAP_CONTENT
+            )
+            layoutParams.topMargin =
+                context.resources.getDimensionPixelSize(R.dimen.question_text_padding)
+            layoutParams.bottomMargin =
+                context.resources.getDimensionPixelSize(R.dimen.question_text_padding)
+            this.layoutParams = layoutParams
+
+            background = ContextCompat.getDrawable(context, R.drawable.main_button_background)
+        }
+        selectedDateTimeLabel = TextView(context).apply {
+            text = "$selectedDate $selectedTime"
+            textSize = context.resources.getDimension(R.dimen.date_time_label_text_size)
+            setTextColor(ContextCompat.getColor(context, R.color.black))
+            textAlignment = AppCompatTextView.TEXT_ALIGNMENT_CENTER
+            gravity = Gravity.CENTER
+            val layoutParams = LayoutParams(
+                LayoutParams.MATCH_PARENT,
+                LayoutParams.WRAP_CONTENT
+            )
+            this.layoutParams = layoutParams
+        }
+
+        addView(selectedDateTimeLabel)
+        addView(showDatePickerButton)
+        addView(showTimePickerButton)
+    }
+
+    @Parcelize
+    data class SelectedTime(val hour: Int, val minute: Int) : Parcelable {
+        override fun toString(): String {
+            return "$hour:$minute"
+        }
+    }
+
+    @Parcelize
+    data class SelectedDate(val year: Int, val month: Int, val dayOfMonth: Int) : Parcelable {
+        override fun toString(): String {
+            return "$year/${month + 1}/$dayOfMonth"
+        }
+    }
+
+    override fun style(surveyTheme: SurveyTheme) {
+        showDatePickerButton.setTextColor(surveyTheme.themeColor)
+        showTimePickerButton.setTextColor(surveyTheme.themeColor)
+
+        showDatePickerButton.background.setTint(surveyTheme.themeColor)
+        showTimePickerButton.background.setTint(surveyTheme.themeColor)
+    }
+
+    override fun onTimeSet(view: TimePicker, hourOfDay: Int, minute: Int) {
+        selectedTime = SelectedTime(hourOfDay, minute)
+        selectedDateTimeLabel.text = "$selectedDate $selectedTime"
+    }
+
+    override fun onDateSet(view: DatePicker, year: Int, month: Int, dayOfMonth: Int) {
+        selectedDate = SelectedDate(year, month, dayOfMonth)
+        selectedDateTimeLabel.text = "$selectedDate $selectedTime"
+    }
+}
```

Comment:
Please add a new line at the end ( https://unix.stackexchange.com/questions/18743/whats-the-point-in-adding-a-new-line-to-the-end-of-a-file )